### PR TITLE
Allow previewing unsaved edits of draft posts on self-hosted sites

### LINKED ISSUE #16287: Post Preview Doesn't Work When Editing an Uploaded Draft Post on Self-Hosted Sites
### Expected behavior
It should be possible to preview your changes to a draft post on a self-hosted site.

### Actual behavior
The preview works on a new post that is uploaded for the preview. Changing that post and attempting to preview it shows a preview not possible screen.

<details>
<summary>Empty Preview Screenshot</summary>

image

Note that this screenshot was taken with dark mode enabled, and there is [a bug](https://github.com/wordpress-mobile/WordPress-Android/issues/12076) in the empty preview screen preventing the "Preview Unavailable" text from being displayed when the device is in dark mode.
</details>

### Steps to reproduce the behavior
1. Using a self-hosted site that is not connected to Jetpack, create a new post
2. Preview (overflow menu -> Preview)
3. Observe the preview is properly displayed
4. Exit the preview
5. Change the post content (do not save the post)
6. Preview the Post
7. Observe that the preview fails to load

### PR DESCRIPTION
**Pull request by mchowning**
Fixes #16287. We cannot view previews of scheduled posts or edits to published posts on self-hosted sites (results in an error). We can, however, view previews of edits to draft posts, but it was not possible to do that in the app. This PR updates the app to allow uses to view previews of edited draft posts on self-hosted sites.

In an [earlier PR](https://github.com/wordpress-mobile/WordPress-Android/pull/10322) , this logic was added to disallow previews on self-hosted sites. The test steps for that PR though, indicate that the preview should be available if you "Create (or edit) a draft." Although the "Create" case worked after that PR, I don't think the "or edit" case was, and that is what this PR is addressing.

### New post -> Preview -> Edit -> Preview
1. Using a self-hosted site that is not connected to Jetpack, create a new post
2. Preview (overflow menu -> Preview)
3. Observe the preview is properly displayed
4. Exit the preview
5. Change the post content (do not save the post)
6. Preview the Post
7. Observe that an empty preview is loaded

<details>
<summary>Empty Preview Screenshot</summary>

image

_Note that this screenshot was taken with dark mode enabled, and there is [a bug](https://github.com/wordpress-mobile/WordPress-Android/issues/12076) in the empty preview screen preventing the "Preview Unavailable" text from being displayed when the device is in dark mode._
</details>

### Posts that should still show an empty preview
On a self-hosted site that is not connected to Jetpack, we should not be able to view previews (i.e., the empty preview screen should show). Verify that the empty preview is shown instead of an error dialog appearing because the preview failed:
1. Published posts that have local changes
2. Scheduled posts

## Regression Notes
1. Potential unintended areas of impact
None other than those noted in the test steps.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Went through the test steps in this PR and https://github.com/wordpress-mobile/WordPress-Android/pull/10322

3. What automated tests I added (or what prevented me from doing so)
Added one unit test verifying the new behavior.

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by mchowning** [comment]
:wave: @develric ! I added you as a reviewer only because I saw that you had been poking around the preview logic before. Feel free to let me know if you don't have time to take a look at this or if there is someone else who might be better. There is no rush on reviewing this right away, it's not urgent.

**Comment by develric** [comment]
Hey @mchowning 👋 
>I added you as a reviewer only because I saw that you had been poking around the preview logic before

Happy to help here 🙇 

>There is no rush on reviewing this right away, it's not urgent.

👍. As shared on slack I'll try to come to it tomorrow or possibly on Tuesday after upcoming AFK 🙇

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 *** PLEASE FOLLOW THIS FORMAT: [<priority indicator, more stars = higher priority>] <description> [<PR URL>]
 
+* [*] On self-hosted sites, allow previewng unsaved changes to draft posts [https://github.com/wordpress-mobile/WordPress-Android/pull/16296]
```
Comment (develric) [inline RELEASE-NOTES.txt]: np: small typo previewng vs previewing

**Review by develric — APPROVED** [review]
Hey @mchowning 👋 , thanks for this PR. Code LGTM and I tested as per steps without any issue 👍 ! I only left a np about what I think is a small typo. Approving, feel free to merge it after you have considered the np 🙇

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (mchowning) [reply RELEASE-NOTES.txt]: Thanks for catching that! 🙇  Addressed in [df4d1fa](https://github.com/wordpress-mobile/WordPress-Android/pull/16296/commits/df4d1fa0709b4f16d3dd1b3a27fa5e884f59f1f0)

**Pull request merged**